### PR TITLE
feat(mantine, read-only): add style to form components

### PR DESCRIPTION
**Pull request by GermainBergeron**
### Proposed Changes

Added new style to handle the `readOnly` state on most input components.
Created a [new page](https://plasma.coveo.com/feature/feat-ADUI-10204-read-only-fields/form/Form) to demo the different state of inputs
Added a few demos

### Potential Breaking Changes

None

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/mantine/src/components/read-only/PasswordInput.tsx
Comment (gdostie) [inline packages/mantine/src/components/read-only/PasswordInput.tsx:1]: The password input still has the hand cursor when hovering over the input in read only. I don't know if it is expected?

**Review by gdostie — APPROVED** [review]
Nicely done!

File: packages/mantine/src/components/read-only/PasswordInput.tsx
Comment (gdostie) [reply packages/mantine/src/components/read-only/PasswordInput.tsx:1]: Woops not the password input, the FileInput rather

Comment (GermainBergeron) [reply packages/mantine/src/components/read-only/PasswordInput.tsx:1]: Will use a `cursor: text` instead 👍

**Pull request merged**